app/gist/controller.js

```diff
@@ -60,7 +66,9 @@ export default Em.Controller.extend({
   }),
 
   rebuildApp: Em.observer('model.files.@each.content', function() {
-    Em.run.debounce(this, this.buildApp, 500);
+    if (this.get('isAutorun')) {
```

Comment:
you will also need to watch `isAutoRun` here I think so it runs when someone re-enables autorun.